[SYCL] Ambiguous reference to `PointerType`

### Describe the bug

563401d956cd33d845618cecc1ef1ba32d51e0ec adds a new definition for `PointerType` in `llvm/include/llvm/SYCLLowerIR/MangleUtils.h` that conflicts with the one in `llvm/include/llvm/IR/InlineAsm.h`. Although the two definitions are in different namespaces, the `MangleUtils.cpp` file has using directives for both namespaces.

There's no error when compiling with GCC or with pre-compiled headers disabled since `MangleUtils.cpp` does not include `InlineAsm.h`. The error only occurs when compiling with pre-compiled headers enabled since they include the conflicting definition.

### To reproduce

Compile using Clang with pre-compiled headers enabled. Compilation should produce the following diagnostic:
```
llvm/lib/SYCLLowerIR/MangleUtils.cpp:100:6: error: reference to 'PointerType' is ambiguous
```


### Environment

- OS: Linux
- Target device and vendor: NA
- DPC++ version: 563401d956cd33d845618cecc1ef1ba32d51e0ec
- Dependencies version: NA


### Additional context

_No response_

--- a/llvm/lib/SYCLLowerIR/MangleUtils.cpp
+++ b/llvm/lib/SYCLLowerIR/MangleUtils.cpp
@@ -15,8 +15,8 @@
 #include "llvm/Support/MathExtras.h"
 #include "llvm/Support/raw_ostream.h"
 
-using namespace llvm;
-using namespace llvm::SPIR;
+namespace llvm {
+namespace SPIR {
 
 namespace {
 
@@ -291,3 +291,6 @@ void NameMangler::mangle(StringRef Name, ArrayRef<RefParamType> Params,
   for (const auto &P : Params)
     P->accept(&Visitor);
 }
+
+} // namespace SPIR
+} // namespace llvm

